### PR TITLE
B003 backend fix 

### PR DESCRIPTION
**Pull request by rwedzony**
Fixed backend

### CODE REVIEW

File: src/main/resources/db/migration/V1__create_user_table.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+CREATE TABLE APPUSER
+(
+    id serial,
+    first_name varchar(255),
+    last_name  varchar(255),
+    phoneNumber  varchar(255),
```
Comment (staniakm) [inline src/main/resources/db/migration/V1__create_user_table.sql:7]: 255 for phone number ?

File: src/main/resources/db/migration/V1__create_user_table.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    address  varchar(255),
+    city  varchar(255),
+    postcode  varchar(255),
```
Comment (staniakm) [inline src/main/resources/db/migration/V1__create_user_table.sql:10]: 255 for postcode ?

**Review by staniakm — COMMENTED** [review]
Some issues must be fixeed before merge

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
 spring.jpa.database=POSTGRESQL
-spring.jpa.generate-ddl=true
+spring.jpa.generate-ddl=false
 spring.jpa.hibernate.ddl-auto=none
-spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation=true
-spring.jpa.show-sql=true
+
+
+
+#flyway properties
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/resources/application.properties]: removed

File: src/main/resources/db/migration/afterMigrate.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
+--User insertions
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/resources/db/migration/afterMigrate.sql]: moved

File: src/main/java/it/hospital/project/flyway/FlywayConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.stereotype.Component;
+
+@Component
+public class FlywayConfiguration {
+
+    @Bean
+    public FlywayMigrationStrategy cleanMigrateStrategy() {
+
+        return flyway -> {
+            flyway.clean();
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/java/it/hospital/project/flyway/FlywayConfiguration.java:14]: I fully agree. Thats the reason why I included clean. Without clean instruction, when you add one migration, to test something, start the app,  than  remove it (migration) (without impacting of course previous ones) and start app again, your database schema won't be dropped, but stayes how it was, you only get message: "current schema is ahead one migration".

File: src/main/java/it/hospital/project/appusers/User.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+
+@Data
+@NoArgsConstructor
+@Entity
+@Table(name = "Appuser")
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/java/it/hospital/project/appusers/User.java]: of course

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -1,12 +1,29 @@
 server.port=9090
+
+
+#postgresql database
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/resources/application.properties:4]: I did some clean up

File: src/main/resources/db/migration/V1__create_user_table.sql [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    email  varchar(255),
+    password varchar(255),
+    PRIMARY KEY(id)
+)
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/resources/db/migration/V1__create_user_table.sql]: 100% true

File: pom.xml [hunk 9]
```diff
@@ -48,6 +46,19 @@
 			<artifactId>postgresql</artifactId>
 			<scope>runtime</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>org.flywaydb</groupId>
+			<artifactId>flyway-core</artifactId>
+			<version>6.5.0</version>
```
[the remark this answers is not in the document]
Comment (rwedzony) [reply pom.xml]: done

File: src/main/resources/db/migration/V1__create_user_table.sql (hunk 1 above)
Comment (rwedzony) [reply src/main/resources/db/migration/V1__create_user_table.sql]: consequences of solving to many issues at the time :)

File: src/main/resources/db/migration/V1__create_user_table.sql (hunk 2 above)
Comment (rwedzony) [reply src/main/resources/db/migration/V1__create_user_table.sql]: the same as above

File: src/main/java/it/hospital/project/flyway/FlywayConfiguration.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (staniakm) [reply src/main/java/it/hospital/project/flyway/FlywayConfiguration.java:14]: Clean option drop db before each application start, so it is impossible to store any data between restarts.

[the remark this answers is not in the document]
Comment (rwedzony) [reply src/main/java/it/hospital/project/flyway/FlywayConfiguration.java]: solved by removing this flyway strategy and adding flyway maven plugin to clean database manually

**Review by staniakm — APPROVED** [review]

**Pull request merged**